test/Datadog.Trace.Tests/Propagators/B3SpanContextPropagatorTests.cs

```diff
@@ -0,0 +1,201 @@
+using System.Globalization;
+using System.Linq;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Propagation;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Propagators
+{
+    public class B3SpanContextPropagatorTests : HeadersCollectionTestBase
+    {
+        private B3SpanContextPropagator _propagator;
+
+        public B3SpanContextPropagatorTests()
+        {
+            _propagator = new B3SpanContextPropagator();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetHeaderCollectionImplementations))]
+        internal void HttpRequestMessage_InjectExtract_Identity(IHeadersCollection headers)
```

Comment:
I recommend removing the initial `HttpRequestMessage` and `WebRequest` prefixes since the test methods are testing multiple implementations